src/KOKKOS/kokkos_type.h

```diff
@@ -1076,15 +1076,29 @@ struct params_lj_coul {
 
 // Pair SNAP
 
+#define SNAP_KOKKOS_REAL double
+#define SNAP_KOKKOS_HOST_VECLEN 1
+
+#ifdef LMP_KOKKOS_GPU
+#define SNAP_KOKKOS_DEVICE_VECLEN 32
+#else
+#define SNAP_KOKKOS_DEVICE_VECLEN 1
+#endif
+
+
+// intentional: SNAreal/complex gets reused beyond SNAP
 typedef double SNAreal;
 
 //typedef struct { SNAreal re, im; } SNAcomplex;
-template <typename real>
-struct alignas(2*sizeof(real)) SNAComplex
+template <typename real_>
```

Comment:
Can we change `real` to `real_type`? Also why the underscore?

Reply:
Will do!

Reply:
Done here and elsewhere. I use `int vector_length_`  a few places, I didn't touch those, couldn't think of a good clarifying name. Those instances are pretty closely followed by a `static constexpr int vector_length = vector_length_;` anywho.

Reply:
Oh, I may have misunderstood your request. Do you mean replacing `real` with `real_type` everywhere, and not my specific change of `real_` -> `real_type`?

Reply:
>Do you mean replacing real with real_type everywhere

Yes, that was my thought. I'm open to other suggestions if you have a preference.

Reply:
No preference, `real_type` is growing on me. I'll take care of it later today.